### PR TITLE
feat: bootstrap skeleton

### PR DESCRIPTION
**Pull request by joshuarli**
skeleton for bootstrap, including interactively configuring coderoot.

Will be following up with individual portions of bootstrap after; trying to break this up into smaller reviewable PRs.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -71,18 +76,36 @@ def main(argv: Sequence[str] | None = None) -> int:
...
+    coderoot = config["devenv"]["coderoot"]
+
+    if args.command == "bootstrap":
+        return bootstrap.main(coderoot, remainder)
+
+    # the remaining tools are repo-specific
     if not args.nocoderoot and not args.pwd.startswith(os.path.expanduser(f"~/{coderoot}")):
         print(
-            f"You aren't in your code root (~/{coderoot})!"
+            f"You aren't in your code root (~/{coderoot})! "
```
Comment (IanWoodard) [inline main.py:108]: Is there a reason for the extra space here?

File: main.py [hunk 2]
```diff
@@ -71,18 +76,36 @@ def main(argv: Sequence[str] | None = None) -> int:
...
+
+        os.makedirs(config_root, exist_ok=True)
+        with open(config_path, "w") as f:
+            config.write(f)
+        print(f"If you made a mistake, you can edit {config_path}.")
 
-    # TODO: read a well-known json for preferences
-    coderoot = "dev"
+    config.read(config_path)
+    coderoot = config["devenv"]["coderoot"]
```
Comment (IanWoodard) [inline main.py:100]: We might want to check that the coderoot actually exists

**Review by IanWoodard — COMMENTED** [review]
Looking good so far

File: main.py (hunk 1 above)
Comment (joshuarli) [reply main.py:108]: Yep, it's concatenated with the string below.

File: main.py (hunk 2 above)
Comment (joshuarli) [reply main.py:100]: It doesn't have to exist for the bootstrap step (as bootstrap will be creating the coderoot if it doesn't exist)
and the subsequent `args.pwd.startswith(coderoot)` will fail if coderoot doesn't exist

**Review by IanWoodard — APPROVED** [review]

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -24,10 +24,19 @@ jobs:
           # should be able to be run from anywhere
           ./install-devenv.sh "${{ github.event.pull_request && github.head_ref || github.ref_name }}"
 
+      - name: bootstrap
+        shell: zsh -i {0}
+        run: |
+          mkdir -p ~/.config/sentry-devenv
+          cat <<EOF > ~/.config/sentry-devenv/config.ini
+          [devenv]
+          coderoot = /Users/josh/dev
```
[the remark this answers is not in the document]
Comment (joshuarli) [reply .github/workflows/test.yml:33]: whoops - i'll change this to $HOME/dev

**Pull request merged**